### ISSUE TITLE
Won't Install with Tampermonkey (Chrome)

### ISSUE THREAD

**Issue by t-kitsune** [YePpHa/YouTubeCenter#2382]
Chrome won't install it because it thinks it's a security issue. 

This doesn't happen in ViolentMonkey though.

**Comment by ireun** [comment]
Well.. I've just installed the script without any problem, on Chrome, using Tampermonkey..

**Comment by t-kitsune** [comment]
What Chrome Channel are you ussing @ireun? Because it won't install with Tampermonkey on Chrome, Chrome Beta, Chrome Dev and Chrome Canary for me.

**Comment by ireun** [comment]
Stable channel - 56.0.2924.87 (64-bit) and to assure you that I'm not lying.. https://s1.webmshare.com/KEd8M.webm

**Comment by t-kitsune** [comment]
Hmmm. So I guess it's my system only then. Weird.

**Comment by ireun** [comment]
Well, It is weird, and if You'd like to know.. Win 10 here. ;)

**Comment by Piter432** [comment]
@ireun: Strange. Same OS and Chrome channel and the issue occurs on my PC :( I mean, I can't update the userscript no matter if I do it by proper YTC's GitHub page or in Tampermonkey (the userscript loads and after loading, the Tampermonkey tab quickly closes).

I have some flags enabled, but I don't really know which one cause this 😕

**Comment by ireun** [comment]
Hmm, have you tried uninstalling the script first? Also, I don't think if I have any flags enabled..

**Comment by Piter432** [comment]
@ireun: Didn't help. Chrome wants to download this as `.user.js` file...

**Comment by t-kitsune** [comment]
Wait, So it's not just me? A workaround I have done is to use violent monkey by exporting/importing my scripts.

**Comment by Piter432** [comment]
@t-kitsune: You're not alone. However, we don't know how did you configure Chrome on your PC (I think the OS and its version [32-bit or 64-bit] doesn't matter so much).

**Comment by t-kitsune** [comment]
Well for troubleshooting:
OS: Windows 10 Pro
Chrome Channel: Dev
Chrome Version: 58
Flags Enabled: Yes, but only ones that enable material design
------
Again though, I was able to install YouTube Center via Violent Monkey: https://chrome.google.com/webstore/detail/violent-monkey/jinjaccalgkegednnccohejagnlnfdag

If you go to Tampermonkey settings, you can export your scripts to a .zip file and import them into ViolentMonkey.

**Comment by TheEssemCraft** [comment]
I'm having the same problem, here are my specs:
# OS: Windows 8.1 Pro
# Chrome Channel: Stable
# Chrome Version: 56
# Flags Enabled: Only the Material Design ones

**Comment by Piter432** [comment]
I just try to update YouTube+ and Social Fixer userscript version and they're updated fine. Only with YTC there's a problem with update/install, so I guess there must be a bug somewhere in YTC code or Chrome reads some YTC code incorrectly and that's why the Tampermonkey install/update tab closes quickly after loading the whole script in this tab (at least in my case).

**Comment by t-kitsune** [comment]
@piter: I believe this is a tampermonkey bug because as soon as I uninstalled Tampermonkey and Installed Violent Monkey, I was able to install it fine.

**Comment by Piter432** [comment]
@t-kitsune: I don't use Violentmonkey and I don't want to use it, so I'm not sure about this.

**Comment by TheEssemCraft** [comment]
@t-kitsune Yeah, Violent Monkey seems to have more accurate userscript handling than Tampermonkey. I actually made a very simple script a few months back to change an option in a dropdown box, and it worked flawlessly in Violent Monkey; however, every time I used Tampermonkey, it changed the dropdown option, but not the textbox next to it.

**Comment by t-kitsune** [comment]
@Piter432 I never said you had to lol, I just said for the time being, I was able to install it via Violent Monkey.

**Comment by akatreyt** [comment]
what helped me was to hit tempermonkey icon in chrome, create new script, and copying and pasting the script manually then saving. this installs the script for me.

**Comment by CollinChaffin** [comment]
"Chrome won't install it because it thinks it's a security issue".......vague and Chrome doesn't object to TamperMonkey scripts and is not even aware of them since TM is the actual extension and beyond that the scripts are nothing more than that extension's data.  All this thread but not a single screenshot or details of what is even broken.  This current (and any recent past) DEV builds of the YTC user.js script on TM (stable) on Chrome (either stable or DEV) is/has been working flawlessly.

Before you begin posting these "workarounds"  that god forbid others may begin to perform for any similar issues, please provide actual error screenshot specifically of the Chrome "security issue" (no reason you cannot get this) and any capture-able text errors that you feel are preventing TM from running YTC userscript, and please post them here and then we can provide you a real root cause and get this issue closed.

**Comment by CollinChaffin** [comment]
Without your screenshot now that I read it again I would bet anything you had to create a new script because you simply cannot get the script INTO TAMPERMONKEY and when you DOWNLOAD IT with Chrome, you get a warning asking if you want to keep or discard and no matter which you choose, it will not install.  Is that a correct assessment of what is happening and why you then needed to open TM, select "new" script and paste in the YTC script?

**Comment by seanrand** [comment]
Same issue here with Chromium 57.0.2987.133

Other scripts seem to update fine, but trying to update YouTubeCenter triggers this warning:
![screenshot-20170416t124911](https://cloud.githubusercontent.com/assets/929479/25070613/890d7eb4-22a3-11e7-85f2-71be875e6095.png)

CCing @derjanb

**Comment by derjanb** [comment]
@seanrand YouTubeCenter is installing and updating fine here. 
Can you please go to "chrome://extensions/" and check the "Developer mode" checkbox. Then search the Tampermonkey entry and click at the "background.html" link. A new window opens, make sure the network tab is selected (you can also clear the console and the previous network activity if any). And now click at the userscript install link again and post a screenshot of the window. Thanks.

**Comment by seanrand** [comment]
@derjanb, the network activity doesn't look any different than when installing/updating any other userscript. Successful GET with 200 response.

However, I've played around a bit and I'm speculating here, but it might be the size of the script (maybe combined with slow PCs) that's triggering this.

When I'm trying to update YouTubeCenter, Tampermonkey's script update dialog appears for several seconds, however the script info and source are blank, only reinstall/update and cancel buttons are visible. Then the information gets filled in, but the tab closes immediately, the user.js file gets downloaded instead and the "cannot be installed from this website" warning is shown.

When I truncate YouTubeCenter.user.js to 1000 lines, the update prompt works as expected, showing script info and source, without the tab closing and without any Chrom(e/ium) warnings.

**Comment by derjanb** [comment]
@seanrand 
> Tried debugging further, YouTubeCenter updates fine when I dummy out CodeMirror completely in Tampermonkey

Thanks for looking into this. I'll try to fix it at the next beta version.

**Comment by Piter432** [comment]
Issue is fixed both in beta Tampermonkey and newest stable Tampermonkey versions and looks like it was TM's fault (as I thought).
